components/coding/studentPortal/Sidebar.tsx

```diff
@@ -76,6 +79,19 @@ export const Sidebar: React.FC = () => {
 
   const router = useRouter();
 
+  const { loading: totalUserBadgeCountLoading } =
+    useQuery<FetchUserBadgesCountResponse>(FETCH_TOTAL_USER_BADGES_COUNT, {
```

Comment:
I don't want the sidebar to become bloated and become the go-to place to query all sorts of data and update redux.

Let's figure out somewhere else to put this query. Can we put this on the profile page or coaching dashboard page maybe? 

Or can we adapt the initial query that fetches all the enrolled students to calculate total badge as well?

Happy to chat about it

Reply:
@kwiedenhaupt - try making userId in user_coding_badges foreign key to the users table and id column. I think you should be able to create an array relationship between users and the user_coding_badges table, at which point you could adapt the initial query that fetches enrolled students like Vithushan mentioned

I feel like this should work but not 100% sure. Try giving it a shot